build_pipelines/aspnetcorebuild/prepare_project.py

```diff
@@ -0,0 +1,268 @@
+    number from it.
+
+    Returns:
+        The version number for the runtime used by the app.
+    """
+    keys = libraries.keys()
+    for key in keys:
+        if key.startswith(NETCORE_APP_PREFIX):
+            return key[len(NETCORE_APP_PREFIX):]
+    return None
+
+
+def get_runtime_minor_version(deps_path):
+    """Determines the target of the .NET Core runtime needed by the app.
+
+    Reads the given .deps.json file and determines the version of the
+    runtime used by the app.
+
+    Returns:
+        The version of the runtime used by the app.
+    """
+    with open(deps_path, 'r') as src:
+        content = json.load(src)
+        try:
+            libraries = content['libraries']
+            return parse_runtime_minor_version(libraries)
+        except KeyError:
+            return None
+
+
+def get_major_version(version):
+    """Returns the major version of a parsed version.
+
+    Args:
+        version: A string with a .NET Core version.
+
+    Returns:
+        A string with the major version.
+    """
+    parsed_version = version.split('.')
+    return '.'.join(parsed_version[0:2])
+
+
+class BaseImage(object):
+    """The information about the base image for a given .NET Core version."""
+
+    def __init__(self, version, image):
+        """Initializes the BaseImage object.
+
+        Args:
+            version: A string with the .NET Core version.
+            image: A Docker image name
+        """
+        self.version = version
+        self.major_version = get_major_version(self.version)
+        self.image = image
+
+    def supports(self, version):
+        """Determins if this image can run the requested version.
+
+        This method determines if this base image can run the
+        requested .NET Core version. It will do so by:
+        * Comparing version major verison numbers, major versions are
+          not backwards compatible.
+        * Within a major version an older minor version cannot run an
+          app built against a newer minor version.
+
+        Args:
+            version: A string with the requested version to compare.
+
+        Returns:
+            True if the base image can support the requested version,
+            False otherwise.
+        """
+        major_version = get_major_version(version)
+        return (self.major_version == major_version and
+                StrictVersion(self.version) >= StrictVersion(version))
+
+
+def parse_version_map(version_map):
+    """Produces a dictionary of version to Docker tag from the map.
+
+    Parses the given version_map and produces a dictionary that maps
+    all of the supported versions to the Docker images for those
+    versions.
+
+    Returns:
+        The dictionary with the versions and Docker images.
+    """
+    result = []
+    for entry in version_map:
+        try:
+            key, value = entry.split('=')
+            result.append(BaseImage(key, value))
+        except ValueError:
```

Comment:
What will throw this ValueError? I don't think split will and I don't see a 'raise' in the BaseImage constructor.